discovery-provider/src/premium_content/signature.py

```diff
@@ -55,6 +63,7 @@ def get_premium_content_signature(
             cid=args["track_cid"],
             is_premium=args["is_premium"],
             user_wallet=None,
+            user_id=args["user_id"],
```

Comment:
won't this break if the optional `user_id` does not exist in args? we may need to first check that `'user_id' in args`